pennylane/compiler/python_compiler/transforms/measurements_from_samples.py

```diff
@@ -0,0 +1,590 @@
+
+        return sample_op
+
+    @staticmethod
+    def get_postprocessing_func_op_from_block_by_name(
+        block: ir.Block, name: str
+    ) -> func.FuncOp | None:
+        """Return the post-processing FuncOp from the given `block` with the given `name`.
+
+        If the block does not contain a FuncOp with the matching name, returns None.
+
+        Args:
+            block (ir.Block): The MLIR block to search.
+            name (str): The name of the post-processing FuncOp.
+
+        Returns:
+            func.FuncOp: The FuncOp with matching name.
+            None: If no match was found.
+        """
+        for op in block.ops:
+            if isinstance(op, func.FuncOp) and op.sym_name.data == name:
+                return op
+
+        return None
+
+    @classmethod
+    def get_postprocessing_func_from_module_and_insert(
+        cls, postprocessing_module: builtin.ModuleOp, name: str, matched_op: ir.Operation
+    ) -> func.FuncOp:
+        """Get the post-processing FuncOp with the given `name` from the given
+        `postprocessing_module` and inserts it immediately after the FuncOp (in the same block) that
+        contains `matched_op`.
+
+        The post-processing function recovers the original measurement process result from the
+        samples array.
+
+        Args:
+            postprocessing_module (builtin.ModuleOp): The MLIR module containing the post-processing
+                FuncOp.
+            name (str): The name of the post-processing FuncOp.
+            matched_op (Operation): The reference op, the parent of which is used as the
+                reference point when inserting the post-processing FuncOp. This is usually the op
+                matched in the call to match_and_rewrite().
+
+        Returns:
+            func.FuncOp: The inserted post-processing FuncOp.
+        """
+        func_op = matched_op.parent_op()
+
+        assert isinstance(func_op, func.FuncOp), (
+            f"Expected parent of matched op '{matched_op}' to be a func.FuncOp, but got "
+            f"{type(func_op).__name__}"
+        )
+
+        postprocessing_func_op = cls._get_postprocessing_func_op_clone_from_module(
+            postprocessing_module
+        )
+
+        # The function name from jax.jit is 'main'; rename it here
+        postprocessing_func_op.sym_name = builtin.StringAttr(data=name)
+
+        func_op.parent.insert_op_after(postprocessing_func_op, func_op)
+
+        return postprocessing_func_op
+
+    @staticmethod
+    def _get_postprocessing_func_op_clone_from_module(module: builtin.ModuleOp) -> func.FuncOp:
+        """Get a clone of the post-processing FuncOp from the given `module`.
+
+        Helper function to ``_get_postprocessing_func_from_module_and_insert()``.
+
+        This function assumes the module contains only a single op, which is the FuncOp representing
+        the post-processing function. This module typically comes from an inlined xDSL module, e.g.
```

Comment:
Not sure if it is a good question or not: can a jax.jit() block be transformed in the same way as a text block?

Reply:
Yep, thanks @joeycarter! I'm just aware of that after going through the `jax_utils` module.